samples/net/sockets/echo_server/src/tcp.c

```diff
@@ -220,7 +220,7 @@ static int process_tcp(struct data *data)
 			&client_addr_len);
 	if (client < 0) {
 		LOG_ERR("%s accept error (%d)", data->proto, -errno);
-		return -errno;
+		return 0;
```

Comment:
Is this change intended?

Reply:
Yes, Issue noticed with Testsuite. When TLS handshake fails, it results in accept() errors, which causes echo_server to stop. And consecutive tests fails to run. If there is an handshake / accept error echo_server should go back and start listening to accept new incoming connections.

Reply:
Ok, thanks for clarifying, I guess it's fine then.

Reply:
> If there is an handshake / accept error echo_server should go back and start listening to accept new incoming connections.

Just correction/clarification on terminology - it can go back and keep accept'ing new connections. Because indeed, accept() error may be transient (may be not) in the sense of the whole system, i.e. if one accept() fails, next may still succeed. A typical example of this would be no free file descriptor - the accept() has no choice but fail, but next time, some other part of system could free up fd, and accept() would succeed. Error in accept() doesn't change listening status of a socket, i.e. there's not need to call listen() again.

Whether this change *should* be done is a different question. It depends on what's meant by:

> When TLS handshake fails

Why does it fail? Generally, samples are written with "fail fast" approach, to make any failure visible, instead of employing sophisticated (or not) error handling.

Reply:
> samples are written with "fail fast" approach, to make any failure visible,

That's true. But particularly this sample used to test network stack against Testsuite. That's the reason tuning the error handling. Hope it didn't harm.

Reply:
> Why does it fail?

There are some tests, testsuite trying to fail the handshake, expecting TCP connection close. This is fine, but echo_server stopped and unable to continue further.

Reply:
> But particularly this sample used to test network stack against Testsuite. That's the reason tuning the error handling.

Makes sense to me, thanks for explanation.